src/fileimpl.cpp

```diff
@@ -412,7 +412,7 @@ namespace zim
     return cluster;
   }
 
-  offset_t FileImpl::getOffset(const Reader* reader, size_t idx)
+  offset_t FileImpl::getOffset(const Reader* reader, size_t idx) const
```

Comment:
Which state is modified ?

Reply:
This one turns out completely different from `getIndexByTitle` and doesn't even access `FileImpl`'s state. Therefore it can be converted to a static member function.